### DISCUSSION TITLE
Is it possible to bundle used echarts features automatically?

### DISCUSSION THREAD

**Discussion by kingyue737** [ecomfe/vue-echarts#753] [Ideas]
I just saw the awesome new [import code generator](https://vue-echarts.dev/#codegen). Is it possible to bundle the necessary ECharts components and generate the correct option type via Vite plugin, which can improve DX? 
It may also help split chunks since we usually import all ECharts components in a single module to reduce source code. But if we have two async routes in an SPA, one with a line chart and the other with a bar chart, both ECharts component will be loaded even if we only browse one route. With a bundler plugin, developers no longer need to worry about adding "echarts import code" per route to help split chunks but leave it to plugin or virtual module.
It may be difficult since `option` is an object which is dynamic. Wanna hear other thoughts about this.

**Comment by Justineo** [comment]
> It may be difficult since `option` is an object which is dynamic.

You are right. As the import code can be only generated by the runtime `option` object, it is impossible to safely analyze the dependencies statically at build time. You can only manually use dynamic imports if you want to optimize the overall bundle size.
